public/src/main/java/com/revenuecat/purchases/models/GooglePurchasingData.kt

```diff
@@ -0,0 +1,29 @@
+package com.revenuecat.purchases.models
+
+import com.android.billingclient.api.ProductDetails
+import com.revenuecat.purchases.ProductType
+
+sealed class GooglePurchasingData : PurchasingData {
```

Comment:
So this is currently exposed since it's in the `public` module and with `public` visibility. We should either move it to another module or make it `internal`. 

My suggestion would be to move `PurchasingData` to the `common` module, and this `GooglePurchasingData` to the `feature/google` module. Not certain if there might be some issues with that without trying it though.

Reply:
`GooglePurchasingData` needs to stay where `GoogleStoreProduct` and `GooglePurchaseOption` are.

This is what is inside of `GoogleStoreProduct` 👇 

```kotlin
    override val purchasingData: PurchasingData
        get() = if (type == ProductType.SUBS && defaultOption != null) {
            defaultOption.purchasingData
        } else {
            GooglePurchasingData.InAppProduct(
                productId,
                productDetails
            )
        }
```
If `GooglePurchasingData` is in `feature/google`, then we get a circular dependency 👇 

![Screen Shot 2023-01-20 at 12 20 33 PM](https://user-images.githubusercontent.com/401294/213776846-cac2eb0c-17a2-4dad-a4c9-69361523be73.png)

This is why I move all the Google specific things inside of `feature/google` 🤷‍♂️ 

So I believe we have two options the way it stands right now...
1. `GooglePurchasingInfo` remains in the main module and public
2. All Google specific models move into `feature/google`

@tonidero Thoughts? Or am I mistaken here? 😇

Reply:
Ahhh yes you're completely right... In that case, we probably want to hide the visibility of the `PurchasingInfo` classes within the `public` module instead... We would need to think about it. I think this is ok for the betas but I think we should do this before final release